content/docs/setup/kubernetes/upgrade/steps/index.md

```diff
@@ -36,11 +36,86 @@ are supported. If you are on an older version, please upgrade to 1.1 first.
 [Download the new Istio release](/docs/setup/kubernetes/#downloading-the-release)
 and change directory to the new release directory.
 
+### Istio CNI upgrade
+
+Choose one of the following **mutually exclusive** options to check whether
+[Istio CNI](/docs/setup/kubernetes/additional-setup/cni/) is installed and to update it:
+
+{{< tabset cookie-name="controlplaneupdate" >}}
+{{< tab name="Kubernetes rolling update" cookie-value="k8supdate" >}}
+
+You can use Kubernetes’ rolling update mechanism to upgrade the Istio CNI components.
+This is suitable for cases where `kubectl apply` was used to deploy Istio CNI.
+
+1. To check whether `istio-cni` is installed, search for `istio-cni-node` pods
+   and in which namespace they are running (typically, `kube-system` or `istio-system`):
+
+    {{< text bash >}}
+    $ kubectl get pods -l k8s-app=istio-cni-node --all-namespaces
+    $ NAMESPACE=$(kubectl get pods -l k8s-app=istio-cni-node --all-namespaces --output='jsonpath={.items[0].metadata.namespace}')
+    {{< /text >}}
+
+1. If `istio-cni` is currently installed in a namespace other than `kube-system`
+   (for example, `istio-system`), delete `istio-cni`:
+
+    {{< text bash >}}
+    $ helm template install/kubernetes/helm/istio-cni --name=istio-cni --namespace=$NAMESPACE | kubectl delete -f -
+    {{< /text >}}
+
+1. Install or upgrade `istio-cni` in the `kube-system` namespace:
+
+    {{< text bash >}}
+    $ helm template install/kubernetes/helm/istio-cni --name=istio-cni --namespace=kube-system | kubectl apply -f -
+    {{< /text >}}
```

Comment:
Should we make a note that no new deployment or existing deployment recover/austoscale can work during this window?

Reply:
or this is only impacting the node that istio-cni is being updated?  Should we use some rolling strategy?

Reply:
> Should we make a note that no new deployment or existing deployment recover/austoscale can work during this window?

Well, the real issues are:
* There is at least one race condition, which I'm fixing in https://github.com/istio/cni/pull/118.
* New deployments *will* work, but they may silently not have any iptables redirect configured if they start while the CNI plugin is not installed, i.e. the sidecar will be silently bypassed in this case. And that will not be reported anywhere.

I've been thinking about recommending users to cordon all the nodes where the daemonset is installed, before upgrading. But that might be too much, and it might actually prevent upgrading the daemonset itself.

> or this is only impacting the node that istio-cni is being updated? Should we use some rolling strategy?

`istio-cni-node` being a daemonset, it needs to be upgraded on all nodes. The rolling strategy doesn't matter in this case. When the daemonset pod gets restarted on a given node during rollout, it impacts the creation of pods on that node during that restart. It doesn't matter whether we restart all the daemonset's pods at once or in sequence, since the impact is only local.

Reply:
A better solution I've been thinking about would be to have the Istio CNI scan all pods on the node when it starts, and install or udpate the iptables chains if they are missing or outdated. This will take care of the case when pods are started while the CNI is uninstalled during upgrade.

Reply:
There's nothing to do about this in this particular doc, since the last step is to restart all injected pods, after all Istio components have been upgraded, incl. the Istio CNI.